### ISSUE TITLE
[qlipper] breaks copying from emacs

### ISSUE THREAD

**Issue by RalfJung** [pvanek/qlipper#35]
Steps to reproduce
* Start qlipper
* In emacs, select some text and hit M-w to copy it to the yank ring
* In some other application like Firefox, hit Ctrl-V to paste

Actual behavior
* Nothing happens, or I get the previous content of the clipboard

Expected behavior:
 The text I copied in emacs should be pasted. If I quit qlipper, and do this all again, I see the expected behavior.

Further information:
emacs shows an error when I do M-w while qlipper is running:
> Selection owner couldn't convert: MULTIPLE

I don't know what that means, but maybe it is helpful...

**Comment by ufleisch** [comment]
I can confirm this.
To fix it, you could add in `QlipperItem::QlipperItem(QClipboard::Mode mode)`
```
if (format != "MULTIPLE")
    m_content[format] = mimeData->data(format);
```
However, this could be further improved. There are still useless binary entries in the clipboard history, because Emacs has a lot of strange (probably internal) formats in its clipboard entries ("TIMESTAMP", "MULTIPLE", "text/plain", "COMPOUND_TEXT", "TARGETS", "LENGTH", "DELETE", "FILE_NAME", "CHARACTER_POSITION", "LINE_NUMBER", "COLUMN_NUMBER", "OWNER_OS", "HOST_NAME", "USER", "CLASS", "NAME", "ATOM", "INTEGER", "SAVE_TARGETS"
"text/plain;charset=utf-8" for a simple copied text). Maybe it would be better to restrict the saved content to "real" MIME data, e.g. using something like this:
```
if (format.contains(QLatin1Char('/')))
    m_content[format] = mimeData->data(format);
```
or only store the basic data types provided by `QMimeData` (text(), html(), urls(), imageData(), colorData()).

**Comment by RalfJung** [comment]
That sounds good. Any chance some patch like this could get in? Is qlipper maintained?

**Comment by agaida** [comment]
just make a PR i would suggest :)

**Comment by RalfJung** [comment]
I patched it locally, and will see whether I notice any breakage.

**Comment by ufleisch** [comment]
Just made a PR.

**Comment by palinek** [comment]
fixed in #37
